src/core/src/service_interfaces/StatusHandler.py

```diff
@@ -812,8 +831,157 @@ def __set_errors_json(self, error_count_by_operation, errors_by_operation):
             "details": errors_by_operation,
             "message": message
         }
+
+    def __set_error_detail(self, error_code, message):
+        formatted_message = self.__ensure_error_message_restriction_compliance(message)
+        return {
+            "code": str(error_code),
+            "message": str(formatted_message)
+        }
     # endregion
 
+    # region - Patch Truncation
+    def log_truncated_packages(self):
+        """ log the removed packages from patches in CoreMain after main operation are marked completed """
+        if not len(self.__assessment_packages_removed) == 0:
+            self.composite_logger.log_debug("All packages removed from assessment: [Count={0}] [PackageNames={1}]".format(len(self.__assessment_packages_removed), self.__assessment_packages_removed))
+
+        if not len(self.__installation_packages_removed) == 0:
+            self.composite_logger.log_debug("All packages removed from installation: [Count={0}] [PackageNames={1}]".format(len(self.__installation_packages_removed), self.__installation_packages_removed))
+
+        if len(self.__assessment_packages_removed) == 0 and len(self.__installation_packages_removed) == 0:
+            self.composite_logger.log_debug("No packages truncated")
+
+    def __get_status_payload_with_truncated_patches(self, status_file_payload_json_dumps):
+        """ Get truncated status file payload when status file byte size is more than 126kb """
+        status_file_size_in_bytes = self.__calc_status_size_on_disk(status_file_payload_json_dumps)  # calc complete_status_file_payload_json byte size on disk
+
+        if status_file_size_in_bytes > Constants.StatusTruncationConfig.INTERNAL_FILE_SIZE_LIMIT_IN_BYTES:  # perform truncation complete_status_file byte size > 126kb
+            self.composite_logger.log_debug("Status file limits evaluated. [StatusFileByteSize={0}] [InternalFileByteSize={1}]".format(
```

Comment:
"Status file limits evaluated. [StatusFile**SizeInBytes**={0}] [InternalFile**SizeLimitInBytes**={1}]"

Follow consistent pattern used in rest of the code

Reply:
done